packages/login_client/lib/src/oauth_settings.dart

```diff
@@ -19,10 +19,11 @@ class OAuthSettings {
   /// Creates the [OAuthSettings].
   const OAuthSettings({
     @required this.authorizationUri,
-    this.clientId,
-    this.clientSecret,
+    this.clientId = '',
```

Comment:
`clientId` is required by the spec (and all implementations). Moreover, I don't think it can ever be empty - I haven't seen any OAuth2 servers accepting empty `clientId` (although, AFAIK, spec doesn't disallow that).